pkg/state/state.go

```diff
@@ -47,13 +48,24 @@ func newS3StateConfig() *s3StateConfig {
 	return &s3c
 }
 
-func NewS3State(ctx context.Context, base_path, infix string, client aws.Client) *S3State {
+func NewS3State(ctx context.Context, base_path, infix string, vc *vault.VaultClient, client aws.Client) *S3State {
```

Comment:
As stated above, I don't really like this vault dependency with the AWS code. Maybe another helper class to construct the AWS client using vault? I think about if we had to use another secrets engine...

Reply:
I built that in, cause it enables compatibility with the existing qontract-reconcile code, there we only set up AWS API from vault secrets. I like the idea of having another abstraction for setting up this AWS client, depending on the environment.  Perhaps I can add a CredentialsProvder and use the AWS SDK credentials chain :thinking:

Reply:
Fixed in next commit